grant-project/zcash-grant-system
# SSR & Build Fixes

### What This Does

Fixes a bunch of issues with the production build:

* Fix loadable components not actually working. Even though we were SSRing, it was just throwing out the work and re-rendering due to the loadable chunks not being part of the SSR'd html.
* Removes `dynamic-import-*` babel plugins. These were interfering with loadable components' babel plugin.
* Uncaught SSR errors are now correctly sent through express's error middlewares and rendered. Before they would hang infinitely. We may want to set up a nice 500 page though.
* Adds a `DISABLE_SSL` env var for the production build locally to not force you onto ssl, which didn't work before.
* Sets up server side fetches for requests on `/requests*?` routes.
* Fixed a minor style cascade order bug. Though I'm worried there are more lying in wait. It seems that SSR has a slightly different style order than client rendered, which could be problematic. Let's keep an eye on this.

### Steps to Test

* Update your `.env` with `DISABLE_SSL=true`
* Run `yarn dev` and SSR load every page. Confirm they all load.
* Run `yarn build` and SSR load every page. Confirm they all load.
